app/models/legal_aid_application.rb

```diff
@@ -198,6 +198,52 @@ def has_sca_secure_accommodation_order_proceeding?
     proceedings.any? { |proceeding| proceeding.ccms_code.eql?("PB006") }
   end
 
+  def special_children_act_related_proceedings?
+    proceedings.any? { |proceeding| proceeding.ccms_matter_code.eql?("KPBLW") && proceeding.sca_type == "related" }
+  end
+
+  def client_court_ordered_parental_responsibility?
+    proceedings.any? { |proceeding| proceeding.ccms_matter_code.eql?("KPBLW") && proceeding.relationship_to_child == "court_order" }
+  end
+  alias_method :parental_responsibility_court_order_relationship?, :client_court_ordered_parental_responsibility?
+
+  def client_parental_responsibility_agreement?
+    proceedings.any? { |proceeding| proceeding.ccms_matter_code.eql?("KPBLW") && proceeding.relationship_to_child == "parental_responsibility_agreement" }
+  end
+  alias_method :parental_responsibility_agreement_relationship?, :client_parental_responsibility_agreement?
+
+  def client_biological_parent?
+    proceedings.any? { |proceeding| proceeding.ccms_matter_code.eql?("KPBLW") && proceeding.relationship_to_child.eql?("biological") }
+  end
+  alias_method :biological_parent_relationship?, :client_biological_parent?
+
+  def child_subject_of_proceeding?
+    proceedings.any? do |proceeding|
+      proceeding.ccms_matter_code.eql?("KPBLW") &&
+        (proceeding.relationship_to_child.eql?("child_subject") || proceeding.client_involvement_type_ccms_code == "W")
```

Comment:
I am not sure, just in terms of the application details report, is it possible for providers to select a child-subject client involvement type and a non child-subject relationship to child? Does Dave need to be able to distinguish between these two scenarios?

Reply:
Good question... I think technically yes, and I'll check with Dave on his reporting needs! 😄 

There is a wider issue here that an applicant can be a biological parent of Child 1, have a court order for Child 2 and a responsibility agreement for Child 3 - all of these can be true for a single application.

This scenario is just another wrinkle! 😮 

Another issue is that, currently, on proceeding A - the provider could record they are the biological parent of Child A and on proceeding B, say they have a Court Ordered responsibility for the same child 🤷‍♂️ 
A design ticket is in progress to investigate the parental responsibility question and answer flow